service/src/main/java/org/cbioportal/service/impl/ClinicalAttributeServiceImpl.java

```diff
@@ -52,15 +59,19 @@ public ClinicalAttribute getClinicalAttribute(String studyId, String clinicalAtt
     @PreAuthorize("hasPermission(#studyId, 'CancerStudy', 'read')")
     public List<ClinicalAttribute> getAllClinicalAttributesInStudy(String studyId, String projection, Integer pageSize,
                                                                    Integer pageNumber, String sortBy,
-                                                                   String direction) {
+                                                                   String direction) throws StudyNotFoundException {
+
+        studyService.getStudy(studyId);
```

Comment:
maybe add a comment on top of each place like this? The unaware developer might be tempted to remove it at the first sight, since it looks like it is serving no purpose while the contrary is true (i.e. it is absolutely critical that this line is there).